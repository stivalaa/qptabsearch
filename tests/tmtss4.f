*=======================================================================
* File:    tmtest4.f
* Author:  Alex Stivala
* Created: July 2008
*
* Test program for tmatn.f
*
* $Id: tmtss4.f 2968 2009-11-22 01:21:50Z astivala $
*=======================================================================

      program tmtest4

      implicit none
      
      external tmatn

      integer n1,n2,info
      parameter(n1 = 6, n2 = 13)
      double precision omega1(n1,n1),omega2(n2,n2)
      double precision dmat1(n1,n1),dmat2(n2,n2)
      double precision score
      double precision soln(n1*n2+n1+n2)

*     MATLAB GetTableau('1UBQ')
      data omega1 / 0 , 2.6500 ,  -1.1700 , 2.0600 , -1.2600 ,  -0.5900,
     $     2.6500 ,    0 , 2.1400 ,  -1.4300 , 1.5700 , 2.1000,
     $     -1.1700 , 2.1400 , 1.0000 , 2.0900 ,  -1.0400 ,  -1.2200,
     $     2.0600 ,  -1.4300 , 2.0900 ,    0 , 2.9600 , 2.5800,
     $     -1.2600 , 1.5700 ,  -1.0400 , 2.9600 ,    0 ,  -0.7400,
     $     -0.5900 , 2.1000 ,  -1.2200 , 2.5800 ,  -0.7400 ,    0 /

* pytableaucreate.py -t dssp -n /var/tmp/d1xdtt3.ent
      data omega2 /0.00, -1.11,  1.78, -1.74, -1.65, -2.49,
     $     1.46,  1.89,  1.53, -1.24, -0.84,  2.70,  2.40, 
     $     -1.11,  0.00,  2.89,  1.16, -1.02, -2.59,  1.39,
     $     -0.15,  1.27, -1.70, -0.39,  2.42,  1.96, 
     $     1.78,  2.89,  0.00, -2.06,  2.21, -0.79, -1.72,
     $     0.11, -1.88 , 1.35,  2.55, -0.96, -1.35, 
     $     -1.74,  1.16, -2.06,  0.00,  1.88,  1.95, -0.51,
     $     2.06,  0.32, -2.86, -1.52 , 1.47,  0.86, 
     $     -1.65, -1.02 , 2.21 , 1.88 , 0.00 , 1.66,
     $     2.33 ,-2.15 , 2.15 ,-1.06 ,-0.95 , 1.89 , 2.10 ,
     $     -2.49 ,-2.59 ,-0.79 , 1.95 , 1.66 , 0.00 ,-1.99,
     $     -0.68 , 2.03 ,-1.43 ,-2.60 , 0.48 , 1.10 ,
     $     1.46 , 1.39 ,-1.72 ,-0.51 , 2.33 ,-1.99 , 0.00,
     $     -1.76 ,-0.19 , 2.67 , 1.65 ,-1.54 ,-0.96 ,
     $     1.89 ,-0.15 , 0.11 , 2.06 ,-2.15 ,-0.68 ,-1.76,
     $     0.00 ,-1.91 , 1.36 , 2.62 ,-0.87 ,-1.30 ,
     $     1.53 , 1.27 ,-1.88 , 0.32 , 2.15 , 2.03 ,-0.19,
     $     -1.91 , 0.00 , 2.78 , 1.56 , 1.55 , 0.94 ,
     $     -1.24 ,-1.70 , 1.35 ,-2.86 ,-1.06 ,-1.43 , 2.67,
     $     1.36 , 2.78 , 0.00 ,-1.34 ,-1.91 ,-2.53 ,
     $     -0.84 ,-0.39 , 2.55 ,-1.52 ,-0.95 ,-2.60 , 1.65,
     $     2.62 , 1.56 ,-1.34 , 0.00 , 2.74 , 2.35 ,
     $     2.70 , 2.42 ,-0.96 , 1.47 , 1.89 , 0.48 ,-1.54 ,-0.87,
     $     1.55 ,-1.91 , 2.74 , 0.00 ,-0.62 ,
     $     2.40 , 1.96 ,-1.35 , 0.86 , 2.10 , 1.10 ,-0.96,
     $     -1.30 , 0.94 ,-2.53 , 2.35 ,-0.62 , 0.00 /


      data dmat1 / 0, 0, 0, 0, 0, 0, 
     $             0, 0, 0, 0, 0, 0, 
     $             0, 0, 0, 0, 0, 0, 
     $             0, 0, 0, 0, 0, 0, 
     $             0, 0, 0, 0, 0, 0, 
     $             0, 0, 0, 0, 0, 0  /

      data dmat2 / 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0,
     $             0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0,
     $             0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0,
     $             0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0,
     $             0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0,
     $             0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0,
     $             0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0,
     $             0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0,
     $             0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0,
     $             0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0,
     $             0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0,
     $             0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0,
     $             0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0 /

*     should get score = -7.7909 (from MATLAB TableauMatch tmtest4.m)

      call tmatn(omega1, n1, n1, omega2, n2, n2, .false., .false.,
     $           dmat1, dmat2, 
     $           score, soln, info)
      if (info .ne. 0) then
         write(*,*),'TABMATN FAILED'
      else
         write(*,900),score
 900     format('score = ',f6.2)
      endif
      end

