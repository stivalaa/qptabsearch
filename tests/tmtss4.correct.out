score =  -7.97
